packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -262,7 +264,9 @@ export function usePress(props: PressHookProps): PressResult {
 
     let onKeyUp = (e: KeyboardEvent) => {
       if (state.isPressed && isValidKeyboardEvent(e)) {
-        e.preventDefault();
+        if (shouldPreventDefault(e.target as Element)) {
```

Comment:
I don't like adding this in here, as it's not just the check for type === submit...